pkg/apis/pipeline/v1alpha1/pipelinerun_types.go

```diff
@@ -126,8 +126,11 @@ type PipelineRunStatus struct {
 	// +optional
 	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
 	// map of TaskRun Status with the taskRun name as the key
-	//+optional
+	// +optional
 	TaskRuns map[string]TaskRunStatus `json:"taskRuns,omitempty"`
+	// map of PipelineTask name keys to TaskRun name values
+	// +optional
+	TaskRunNames map[string]string `json:"taskRunNames,omitempty"`
```

Comment:
im kinda tempted to suggest that we find a way to put the names of the `PipelineTasks` insto the existing `taskRuns` status if we can, v.s. having two separate fields for this info, what do you think?

With the current version (copying this from the PipelineRun example in [tutorial.md](https://github.com/knative/build-pipeline/blob/master/docs/tutorial.md#pipeline), we'll end up with something like this:
```yaml
status:
...
  taskRunNames:
      tutorial-pipeline-run-1-build-skaffold-web:
            build-skaffold-web
      tutorial-pipeline-run-1-deploy-web:
            deploy-web
...
  taskRuns:
    tutorial-pipeline-run-1-build-skaffold-web:
      conditions:
        - lastTransitionTime: 2018-12-11T20:31:41Z
          status: "True"
          type: Succeeded
...
```

But I think the interface might be a bit nicer if we put all the related info into `status.taskRuns`, e.g.:

```yaml
status:
...
  taskRuns:
    tutorial-pipeline-run-1-build-skaffold-web:
      pipeline-task-name: build-skaffold-web <--- this would be new
      conditions:
        - lastTransitionTime: 2018-12-11T20:31:41Z
          status: "True"
          type: Succeeded
...
```

Or we could make it something like:

```yaml
status:
...
  taskRuns:
    tutorial-pipeline-run-1-build-skaffold-web:
      pipeline-task-name: build-skaffold-web <--- this would be new
      status: <-- we could put the existing `TaskRunStatus` into a `status` section
        conditions:
          - lastTransitionTime: 2018-12-11T20:31:41Z
            status: "True"
            type: Succeeded
...
```

What do you think?

Reply:
I didn't want to mess with `TaskRunStatus` if I could avoid it - just made things a bit messy.

Reply:
Ooooh. That second option seems cleaner. I'm OK with that approach.

Reply:
@abayer @bobcatfish this would mean the `pipeline-task-name` is only present when the `TaskRun` has been created from a `PipelineRun` right ?

Reply:
Well, when the name has been created, which is basically how it works now.

Reply:
No, I misread - you're right. Lemme see how this all works out. =)